src/cli_shared/cli/mod.rs

```diff
@@ -125,12 +125,15 @@ pub struct CliOpts {
     pub color: LoggingColor,
     /// Turn on tokio-console support for debugging.
     /// Memory leak, see <https://github.com/tokio-rs/console/pull/501>.
+    #[cfg(feature = "tokio-console")]
     #[arg(long)]
     pub tokio_console: bool,
     /// Send telemetry to `grafana loki`
+    #[cfg(feature = "tracing-loki")]
     #[arg(long)]
     pub loki: bool,
     /// Endpoint of `grafana loki`
+    #[cfg(feature = "tracing-loki")]
     #[arg(long, default_value = "http://127.0.0.1:3100")]
     pub loki_endpoint: String,
```

Comment:
The user experience of conditional command-line flags is poor. The flags should be static and sensible errors should be printed if, say, the loki flag is used with enabling `tracing-loki`.

Reply:
Fixed.